mlir/lib/Dialect/AMDGPU/IR/AMDGPUDialect.cpp

```diff
@@ -459,6 +461,29 @@ LogicalResult DPPOp::verify() {
   return success();
 }
 
+LogicalResult GlobalLoadLDSOp::verify() {
+  MemRefType srcType = cast<MemRefType>(getSrc().getType());
+  MemRefType dstType = cast<MemRefType>(getDst().getType());
+
+  if (!memref::isStaticShapeAndContiguousRowMajor(srcType) ||
+      !memref::isStaticShapeAndContiguousRowMajor(dstType))
```

Comment:
... I'm actually not sure it's a hard requirement on the destination either, but I'm more open to buying that